moveit_ros/perception/CMakeLists.txt

```diff
@@ -3,18 +3,25 @@ project(moveit_ros_perception)
 
 add_compile_options(-std=c++11)
 
+option(BUILD_OPENGL "Build the parts that depends on OpenGL" ON)
+
 if(NOT CMAKE_CONFIGURATION_TYPES AND NOT CMAKE_BUILD_TYPE)
   set(CMAKE_BUILD_TYPE Release)
 endif()
 
 find_package(Boost REQUIRED thread signals)
 
-find_package(OpenGL REQUIRED)
-find_package(GLEW REQUIRED)
-find_package(GLUT REQUIRED)
-if (OPENGL_FOUND)
-  set(gl_LIBS ${gl_LIBS} ${OPENGL_LIBRARIES})
-endif(OPENGL_FOUND)
+if(BUILD_OPENGL)
+  find_package(OpenGL REQUIRED)
+  find_package(GLEW REQUIRED)
+  find_package(GLUT REQUIRED)
+  if (OPENGL_FOUND)
+    set(gl_LIBS ${gl_LIBS} ${OPENGL_LIBRARIES})
+  endif(OPENGL_FOUND)
+  set(perception_GL_INCLUDE_DIRS "mesh_filter/include" "depth_image_octomap_updater/include")
+  set(SYSTEM_GL_INCLUDE_DIRS ${GLEW_INCLUDE_DIR} ${GLUT_INCLUDE_DIR})
```

Comment:
this line seems unused, could you please explain?

Reply:
It is used in line 91. If `BUILD_OPENGL` is not set, `SYSTEM_GL_INCLUDE_DIRS` is empty; if `BUILD_OPENGL` is set, the paths are set as `SYSTEM` include directories.